cmd/nerdctl/run_security_linux.go

```diff
@@ -85,21 +89,28 @@ func generateSecurityOpts(securityOptsMap map[string]string) ([]oci.SpecOpts, er
 		}
 	}
 
-	nnp := false
-	if nnpStr, ok := securityOptsMap["no-new-privileges"]; ok {
-		if nnpStr == "" {
-			nnp = true
-		} else {
-			var err error
-			nnp, err = strconv.ParseBool(nnpStr)
-			if err != nil {
-				return nil, fmt.Errorf("invalid \"no-new-privileges\" value: %q: %w", nnpStr, err)
-			}
-		}
+	nnp, err := maputil.MapBoolValueAsOpt(securityOptsMap, "no-new-privileges")
+	if err != nil {
+		return nil, err
 	}
+
 	if !nnp {
 		opts = append(opts, oci.WithNewPrivileges)
 	}
+
+	privilegedWithoutHostDevices, err := maputil.MapBoolValueAsOpt(securityOptsMap, "privileged-without-host-devices")
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Can we return an error when `privilegedWithoutHostDevices && !privileged`

Reply:
Fixed.